python-sdk/pachyderm_sdk/config.py

```diff
@@ -47,6 +94,54 @@ def active_enterprise_context(self) -> "Context":
             raise ConfigError(f"active enterprise context not found: {context_name}")
         return Context(**contexts[context_name])
 
+    def add_context(
+        self, name: str, context: "Context", *, overwrite: bool = False
+    ) -> None:
+        """Add a context to the parsed config.
+        The context name must be unique, unless overwrite is True.
+
+        Parameters
+        ----------
+        name : str
+            The name of the context.
+        context : Context
+            The context object.
+        overwrite : bool (kwarg)
+            Whether to overwrite an existing context.
+        """
+        contexts = self._config_file_data["v2"]["contexts"]
+        if (name in contexts) and not overwrite:
+            raise ValueError(f'context "{name}" already exists')
+        contexts[name] = asdict(context)
+
+    def set_active_context(self, name: str) -> None:
+        """Set the active context to the specified name.
+        The context must exist, else a ValueError is raise.
+
+        Parameters
+        ----------
+        name : str
+            The name of the context.
+        """
+        contexts = self._config_file_data["v2"]["contexts"]
+        if name not in contexts:
+            raise ValueError(f'context "{name}" does not exist')
+        self._config_file_data["v2"]["active_context"] = name
+
+    def write(self, config_file: Union[Path, str]) -> None:
+        """Write the current config data to the specified file.
+
+        Parameters
+        ----------
+        config_file : str
+            The path to the config file.
+        """
+        config_file = Path(os.path.expanduser(config_file)).resolve()
+        data = deepcopy(self._config_file_data)
```

Comment:
for my own edification: why is the deepcopy needed here? we can't just directly `json.dumps` from the config_file_data?

Reply:
umm good question. I definitely had a reason for this when I wrote it but that need must have gotten edited out as I worked. I'll think about it a little more but it should be unnecessary with the current code

Reply:
ended up removing this. Nice catch!